weave/trace/serialize.py

```diff
@@ -9,37 +9,59 @@
     FileCreateReq,
     TraceServerInterface,
 )
+from weave.trace_server.trace_server_interface_util import bytes_digest
 
+if typing.TYPE_CHECKING:
+    from weave.trace.weave_client import WeaveClient
 
-def to_json(obj: Any, project_id: str, server: TraceServerInterface) -> Any:
+
+def to_json(obj: Any, project_id: str, client: "WeaveClient") -> Any:
```

Comment:
Do you need the whole client, or can you get away with just the server?

Reply:
You need the client in particular, so that we can schedule the file creation.

Reply:
Does it make sense to move this method to the client then?  I think the client should have a way to easily get the `project_id`, so you'd basically instead have:
```py
res = client.jsonify(obj)
```